execute/exectypes/costly_messages.go

```diff
@@ -185,3 +202,63 @@ func (n *StaticMessageExecCostUSD18Calculator) MessageExecCostUSD18(
 }
 
 var _ MessageExecCostUSD18Calculator = &StaticMessageExecCostUSD18Calculator{}
+
+// CCIPMessageFeeUSD18Calculator calculates the fees (paid at source) of a set of messages in USD18s.
+type CCIPMessageFeeUSD18Calculator struct {
+	lggr logger.Logger
+
+	ccipReader readerpkg.CCIPReader
+
+	// RelativeBoostPerWaitHour indicates how much to increase (artificially) the fee paid on the source chain per hour
+	// of wait time, such that eventually the fee paid is greater than the execution cost, and we’ll execute it.
+	// For example: if set to 0.5, that means the fee paid is increased by 50% every hour the message has been waiting.
+	relativeBoostPerWaitHour float64
+
+	now func() time.Time
+}
+
+var _ MessageFeeE18USDCalculator = &CCIPMessageFeeUSD18Calculator{}
+
+// MessageFeeUSD18 Returns a map from message ID to the message's fee in USD18s.
+func (c *CCIPMessageFeeUSD18Calculator) MessageFeeUSD18(
+	ctx context.Context,
+	messages []cciptypes.Message,
+	messageTimeStamps map[cciptypes.Bytes32]time.Time,
+) (map[cciptypes.Bytes32]plugintypes.USD18, error) {
+	linkPriceUSD, err := c.ccipReader.LinkPriceUSD(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get LINK price in USD: %w", err)
+	}
+
+	messageFees := make(map[cciptypes.Bytes32]plugintypes.USD18)
+	for _, msg := range messages {
+		feeUSD18 := big.NewInt(0).Mul(linkPriceUSD.Int, msg.FeeValueJuels.Int)
```

Comment:
nit: The big API is a little weird, I thought you were multiply everything by 0. It seems like the authors intended for you to use new, [some examples are in the tests](https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/math/big/int_test.go;l=720).
```suggestion
		feeUSD18 := new(big.Int).Mul(linkPriceUSD.Int, msg.FeeValueJuels.Int)
```

Reply:
Good to know, thanks! I think I copied `big.NewInt(0)` from the v1 Exec plugin.